src/Background/Portal.vala

```diff
@@ -69,40 +80,32 @@ public class Background.Portal : Object {
     ) throws DBusError, IOError {
         debug ("Notify background for '%s'.", app_id);
 
-        uint32 _response = 2;
         var _results = new HashTable<string, Variant> (str_hash, str_equal);
+        uint32 _response = 2;
 
-        var notification_request = new NotificationRequest ();
-
-        notification_request.response.connect ((result) => {
-            switch (result) {
-                case NotificationRequest.NotifyBackgroundResult.CANCELLED:
-                    _response = 1;
-                    break;
-                case NotificationRequest.NotifyBackgroundResult.FAILED:
-                    break;
-                default:
-                    _response = 0;
-                    _results["result"] = (uint32) result;
-                    break;
+        var notification = new NotificationRequest ();
+        notification.response.connect ((result) => {
+            if (result == CANCELLED) {
+                _response = 1;
+            } else if (result != FAILED) {
+                _response = 0;
+                _results["result"] = result;
```

Comment:
`result` isn't interpreted as a `uint` here either
```suggestion
                _results["result"] = (uint32) result;
```
Or change the enum type signature 🤷